python/cudf/bindings/csv.pyx

```diff
@@ -0,0 +1,452 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from .cudf_cpp cimport *
+from .cudf_cpp import *
+from cudf.bindings.csv cimport read_csv, _wrap_string, is_file_like
+from libc.stdlib cimport malloc, free
+from libc.stdint cimport uintptr_t
+from libcpp.vector cimport vector
+
+from cudf.dataframe.column import Column
+from cudf.dataframe.numerical import NumericalColumn
+from cudf.dataframe.dataframe import DataFrame
+from cudf.dataframe.datetime import DatetimeColumn
+from cudf.bindings.nvtx import nvtx_range_push, nvtx_range_pop
+from librmm_cffi import librmm as rmm
+
+import nvstrings
+import numpy as np
+import collections.abc
+import os
+
+cdef char* _wrap_string(text):
+    if(text is None):
+        return <char*>NULL
+    else:
+        output = text.encode()
+        return <char*>output
+        # return ffi.new("char[]", text.encode())
```

Comment:
I'm thinking I should refactor this function out. It only really does what's expected for `None` objects, otherwise if returned to a C object it causes memory lifetime issues, and if returned to a Python object it typecasts it to bytes.